contracts/levels/Shop.sol

```diff
@@ -0,0 +1,19 @@
+pragma solidity 0.4.24;
+
+interface Buyer {
+  function price() external view returns (uint);
+}
+
+contract Shop {
+  uint public price = 100;
+  bool public isSold;
+
+  function buy() public {
+    Buyer _buyer = Buyer(msg.sender);
+
+    if (_buyer.price.gas(3000)() >= price && !isSold) {

```

Comment:
What are you specifying the amount of gas passed? As far as I can tell, the level can be solved completely ignoring gas.

Reply:
Gas specified to prevent modifying the state in Buyer contract.